jwst/flatfield/flat_field.py

```diff
@@ -489,24 +524,24 @@ def nirspec_brightobj(output_model,
                                       dtype=output_model.err.dtype)
     interpolated_flats.wavelength = wl.copy()
 
-    if len(shape) == 3:
-        flat_nd = flat_2d.reshape((1, ysize, xsize))
-        flat_dq_nd = flat_dq_2d.reshape((1, ysize, xsize))
-    else:
-        flat_nd = flat_2d
-        flat_dq_nd = flat_dq_2d
-    output_model.data /= flat_nd
-    output_model.err /= flat_nd
-    output_model.dq |= flat_dq_nd
+    output_model.data /= flat_2d
```

Comment:
BrightObj mode exposures go through the extract_2d step, which I assume would mean they'd also be in the form of a MultiSlitModel, with just 1 slit instance. So why don't they get treated as a slit?

Reply:
That's a good question.  @philhodge any ideas?

The bright object spec2 regression test
```
jwst/tests_nightly/general/nirspec/test_pipelines.py::TestNIRSpecPipelines::test_nrs_fs_brightobj_spec2
```
takes a `CubeModel` rate file and produces a `SlitModel` calints file.
```
In [1]: from jwst import datamodels

In [2]: rateints = datamodels.open('/Users/jdavies/scratch/test_nrs_fs_brightobj_spec20/jw84600042001_02101_00001_nrs2_rateints.fits')

In [3]: rateints
Out[3]: <CubeModel(3, 32, 2048) from jw84600042001_02101_00001_nrs2_rateints.fits>

In [4]: calints = datamodels.open('/Users/jdavies/scratch/test_nrs_fs_brightobj_spec20/jw84600042001_02101_00001_nrs2_calints.fits')

In [5]: calints
Out[5]: <SlitModel(3, 32, 2048) from jw84600042001_02101_00001_nrs2_calints.fits>
```

Reply:
NRS_BRIGHTOBJ data are supposed to be `SlitModel`, not `MultiSlitModel`.  (They used to be `CibeModel`, and it could be that there are some old data of that type in regression tests.)  So the data array is 3-D, with the first axis being integration number, and dividing by a 2-D flat works on the whole array by broadcasting.